deno2/deno.gni

```diff
@@ -0,0 +1,135 @@
+  suffix = "_$name"
+  action("create_snapshot_" + name) {
+    forward_variables_from(invoker,
+                           [
+                             "testonly",
+                             "deps",
+                           ])
+    visibility = [ ":*" ]  # Only targets in this file can depend on this.
+    deps += [ ":snapshot_creator" ]
+    script = "v8/tools/run.py"
+    data = []
+    exe = rebase_path(get_label_info(":snapshot_creator", "root_out_dir") +
+                      "/snapshot_creator")
+    natives_in_bin = "$root_out_dir/natives_blob.bin"
+    snapshot_in_bin = "$root_out_dir/snapshot_blob.bin"
+    natives_out_cc = "$target_gen_dir/natives${suffix}.cc"
+    snapshot_out_cc = "$target_gen_dir/snapshot${suffix}.cc"
+    sources = [
+      invoker.js,
+    ]
+    outputs = [
+      natives_out_cc,
+      snapshot_out_cc,
+    ]
+    args = [
+      exe,
+      rebase_path(invoker.js, root_build_dir),
+      rebase_path(natives_in_bin, root_build_dir),
+      rebase_path(snapshot_in_bin, root_build_dir),
+      rebase_path(natives_out_cc, root_build_dir),
+      rebase_path(snapshot_out_cc, root_build_dir),
+    ]
+
+    # To debug snapshotting problems:
+    #  args += ["--trace-serializer"]
+    data = [
+      invoker.js,
+    ]
+  }
+}
+
+template("rust_crate") {
+  crate_type = invoker.crate_type
+  out_dir = target_out_dir
+  source_root = invoker.source_root
+  if (crate_type == "bin") {
+    output = "$target_out_dir/$target_name"
+    out_dir = root_out_dir
+  } else if (crate_type == "lib") {
+    output = "$target_out_dir/lib$target_name.rlib"
+  } else {
+    assert(false, "Unsupported crate_type")
+  }
+  action(target_name) {
+    script = "v8/tools/run.py"
+    depfile = "$target_gen_dir/$target_name.d"
+    sources = [ source_root ]
+    outputs = [ output ]
+    forward_variables_from(invoker, ["deps"])
+    args = [
+      "rustc",
+      rebase_path(source_root, root_build_dir),
+      "--crate-name=$target_name",
+      "--emit=dep-info,link",
+      "--crate-type=$crate_type",
+      "--cfg",
+      "feature=\"default\"",
+      "--cfg", 
+      "feature=\"use_std\"",
+      "--out-dir",
+      rebase_path(out_dir, root_build_dir),
+      "-L",
+      "dependency=" + rebase_path(depfile, root_build_dir),
+    ]
+    if (!defined(deps)) {
+      deps = []
+    }
+    #   --extern libc=$out/liblibc.rlib \
+    foreach (dep_label, deps) {
+      dep_name = get_label_info(dep_label, "name")
+      dep_dir = get_label_info(dep_label, "target_out_dir")
+      args += [
+        "--extern",
+        "$dep_name=" + rebase_path("$dep_dir/lib$dep_name.rlib"),
+      ]
+    }
+    if (defined(invoker.non_rust_deps)) {
+      deps += invoker.non_rust_deps
+      foreach (dep_label, deps) {
+        dep_dir = get_label_info(dep_label, "target_out_dir")
+        args += [
+          "-L",
+          rebase_path(dep_dir),
+        ]
+      }
+    }
+    if (is_debug) {
+      args += ["-C", "debuginfo=2"]
```

Comment:
You probably also want `args += ["-C", "opt-level=3"]` for release mode.

Reply:
Thanks - added.